cobalt/browser/h5vcc_experiments/h5vcc_experiments_impl.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/browser/h5vcc_experiments/h5vcc_experiments_impl.h"
+
+#include "base/feature_list.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/metrics/field_trial_params.h"
+#include "build/build_config.h"
+#include "cobalt/browser/cobalt_experiment_names.h"
+#include "cobalt/browser/global_features.h"
+#include "components/prefs/pref_service.h"
+
+namespace h5vcc_experiments {
+
+namespace {
+
+mojom::OverrideState GetFeatureInternal(
+    std::map<std::string, h5vcc_experiments::mojom::OverrideState>&
+        override_map,
+    const std::string& feature_name) {
+  // auto accessor = cobalt::GlobalFeatures::GetInstance()->accessor();
+  // base::FeatureList::OverrideState override =
+  //     accessor->GetOverrideStateByFeatureName(
+  //         feature_name);
+  // if (override == base::FeatureList::OverrideState::OVERRIDE_ENABLE_FEATURE)
+  // {
+  //   return h5vcc_experiments::mojom::OverrideState::OVERRIDE_ENABLE_FEATURE;
+  // } else if (override ==
+  // base::FeatureList::OverrideState::OVERRIDE_DISABLE_FEATURE) {
+  //   return h5vcc_experiments::mojom::OverrideState::OVERRIDE_DISABLE_FEATURE;
+  // }
+  // return h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+
+  // Cache feature params in a map for faster lookup.
```

Comment:
I thought caching feature param results is slightly better than using accessor even though it requires manual parsing of the features. Lmk what you think @joeltine

Reply:
Caching seems strictly better, but is GetOverrideStateByFeatureName not already using its own internal cache? I'd be surprised if it was doing something expensive under the hood. Also, if needed, can't we generate this cache earlier in the browser process? E.g., immediately after loading the experiment config? 

Also, let's be careful with the names "feature params" vs "features". In this case, I think you're looking up "Features".